homeassistant/components/foscam/camera.py

```diff
@@ -85,83 +78,90 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a Foscam IP Camera."""
+    LOGGER.warning(
+        "Loading foscam via platform config is deprecated, it will be automatically imported. Please remove it afterwards."
+    )
+
+    config_new = {
+        CONF_NAME: config[CONF_NAME],
+        CONF_HOST: config["ip"],
+        CONF_PORT: config[CONF_PORT],
+        CONF_USERNAME: config[CONF_USERNAME],
+        CONF_PASSWORD: config[CONF_PASSWORD],
+        CONF_STREAM: "Main",
+    }
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config_new
+        )
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a Foscam IP camera from a config entry."""
     platform = entity_platform.current_platform.get()
-    assert platform is not None
     platform.async_register_entity_service(
-        "ptz",
-        {
-            vol.Required(ATTR_MOVEMENT): vol.In(
-                [
-                    DIR_UP,
-                    DIR_DOWN,
-                    DIR_LEFT,
-                    DIR_RIGHT,
-                    DIR_TOPLEFT,
-                    DIR_TOPRIGHT,
-                    DIR_BOTTOMLEFT,
-                    DIR_BOTTOMRIGHT,
-                ]
-            ),
-            vol.Optional(ATTR_TRAVELTIME, default=DEFAULT_TRAVELTIME): cv.small_float,
-        },
-        "async_perform_ptz",
+        SERVICE_PTZ, SERVICE_PTZ_SCHEMA, "async_perform_ptz"
```

Comment:
We shouldn't use that service schema as the entity service helper generates an entity service schema for us if we pass a dict like we did before this PR.

Reply:
Ok, will revert this change then. I thought it was required to do it like this now

Reply:
New PR: https://github.com/home-assistant/core/pull/45154